src/engine/enginebuffer.cpp

```diff
@@ -1034,83 +1031,30 @@ void EngineBuffer::process(CSAMPLE* pOutput, const int iBufferSize) {
         // really care about. It will try very hard to keep these in memory
         hintReader(rate);
-    }
 
-    const double kSmallRate = 0.005;
-    if (m_bLastBufferPaused && !bCurBufferPaused) {
-        if (fabs(rate) > kSmallRate) { //at very slow forward rates, don't ramp up
-            m_iRampState = ENGINE_RAMP_UP;
-        }
-    } else if (!m_bLastBufferPaused && bCurBufferPaused) {
-        m_iRampState = ENGINE_RAMP_DOWN;
-    } else { //we are not changing state
-        // Make sure we aren't accidentally ramping down. This is how we make
-        // sure that ramp value will become 1.0 eventually.
-        //
-        // 9/2012 rryan -- As I understand it this code intends to prevent us
-        // from getting stuck ramped down. If there is a meaningfully large rate
-        // and we aren't ramped up completely then it makes us ramp up. This
-        // causes crazy feedback if you scratch at the non-silent end of a
-        // track. See Bug #1006111. I added a !bCurBufferPaused term here because
-        // if rate > 0 and bCurBufferPaused then basically you are at the end of
-        // the track and trying to jog forward so this uniquely blocks that
-        // situation.
-        if (fabs(rate) > kSmallRate && !bCurBufferPaused &&
-            m_iRampState != ENGINE_RAMP_UP && m_fRampValue < 1.0) {
-            m_iRampState = ENGINE_RAMP_UP;
-        }
-    }
+        // release the pauselock
+        m_pause.unlock();
+    } else { // if (!bTrackLoading && m_pause.tryLock()) {
+        // We are loading a new Track
+        bCurBufferPaused = true;
 
-    // let's try holding the last sample value constant, and pull it
-    // towards zero
-    float ramp_inc = 0;
-    if (m_iRampState == ENGINE_RAMP_UP ||
-        m_iRampState == ENGINE_RAMP_DOWN) {
-        // Ramp of 3.33 ms
-        ramp_inc = static_cast<double>(m_iRampState * 300) / sample_rate;
-
-        for (int i=0; i < iBufferSize; i += 2) {
-            if (bCurBufferPaused && !m_bCrossfadeReady) {
-                //qDebug() << "ramp dither";
-                CSAMPLE dither = m_pDitherBuffer[m_iDitherBufferReadIndex];
-                m_iDitherBufferReadIndex = (m_iDitherBufferReadIndex + 1) % MAX_BUFFER_LEN;
-                pOutput[i] = m_fLastSampleValue[0] * m_fRampValue + dither;
-                pOutput[i+1] = m_fLastSampleValue[1] * m_fRampValue + dither;
-            } else {
-                //qDebug() << "ramp buffer";
-                pOutput[i] = pOutput[i] * m_fRampValue;
-                pOutput[i+1] = pOutput[i+1] * m_fRampValue;
-            }
+        // Here the old track was playing and loading the new track is in
+        // progress so we have not the chance to collect real samples for
+        // fade out.
+        // We apply a rectangular Gain change here which may click.
+        // It is in the responds of the user to not play this to
+        // the audience.
+        //if (m_speed_old) {
```

Comment:
We can't predict when it happens. So we are not able to collect old samples. New samples are also not in place and we can't predict when they will be in place.  

This case happens only if it is allowed to load a track into a playing track. If one does this, a click from breaking the last track is somehow natural and he should know that such sound should not be played to the master (audiance). 
Simply pause the track before. 

A click free solution requires more refectory, which is out of the scope of this PR.  
Fading the last sample to zero clicks too.